### PR TITLE
Feature/config store

### PR DESCRIPTION
**Pull request by kmonson**
Agent side of the config store.

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/736)

### CODE REVIEW

File: volttron/platform/store.py [hunk 1]
```diff
@@ -284,9 +233,40 @@ def set_config(self, config_name, contents, trigger_callback=False):
...
+
+        #We need to create store and lock if it doesn't exist in case someone tries to add
+        # a configuration while we are sending the initial state.
+        agent_store = self.store.get(identity)
+
+
+        if agent_store is None:
+            # Initialize a new store.
+            store_path = os.path.join(self.store_path, identity + store_ext)
+            store = PersistentDict(filename=store_path, flag='c', format='json')
```
Comment (craig8) [inline volttron/platform/store.py]: What will happen if someone puts an invalid path say for instanc e a <, > character will your store path blow up?

File: volttron/platform/vip/agent/subsystems/configstore.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+from .base import SubsystemBase
+from volttron.platform.storeutils import list_unique_links, check_for_config_link
+
+from collections import defaultdict
+from copy import deepcopy
+
+"""The heartbeat subsystem adds an optional periodic publish to all agents.
+Heartbeats can be started with agents and toggled on and off at runtime.
+"""
+
```
Comment (craig8) [inline volttron/platform/vip/agent/subsystems/configstore.py]: I don't understand why this comment is here.

Can you add some documentation to this file.  I know most of the methods are documented via the spec, however that doesn't get added to our api documentation output.

One more thing of note, should we have "public" methods first in the file or second...I know it really doesn't make a difference, but thought I would ask your opinion

**Comment by craig8** [comment]
@kmonson can you address these please.  I will get it downloaded and try to implement one of the stores and then go ahead and merge it in.

File: volttron/platform/vip/agent/subsystems/configstore.py (hunk 2 above)
Comment (kmonson) [inline volttron/platform/vip/agent/subsystems/configstore.py]: Bad copy/paste... 

I like to put my helper functions at the top as I usually write them first.

I can add some doc strings.

File: volttron/platform/store.py (hunk 1 above)
Comment (kmonson) [inline volttron/platform/store.py]: Yeah, right now it will probably blowup. Should only affect the bad agent and not cause any lasting harm.

**Pull request merged**